src/go/rdctl/pkg/shutdown/shutdown.go

```diff
@@ -54,72 +60,51 @@ func FinishShutdown(waitForShutdown bool) error {
 	return nil
 }
 
-func doCheckWithTimeout(checkFunc func() bool, killFunc func() error, waitForShutdown bool, retryCount int, retryWait int, operation string) error {
-	for iter := 0; waitForShutdown && iter < retryCount; iter++ {
+func (s *ShutdownInfo) waitForAppToDieOrKillIt(checkFunc func() (bool, error), killFunc func() error, retryCount int, retryWait int, operation string) error {
+	for iter := 0; s.waitForShutdown && iter < retryCount; iter++ {
 		if iter > 0 {
 			logrus.Debugf("checking %s showed it's still running; sleeping %d seconds\n", operation, retryWait)
 			time.Sleep(time.Duration(retryWait) * time.Second)
 		}
-		if !checkFunc() {
+		status, err := checkFunc()
+		if err != nil {
+			return fmt.Errorf("checking operation %s => error %w", operation, err)
+		}
+		if !status {
 			logrus.Debugf("%s is no longer running\n", operation)
 			return nil
 		}
 	}
 	logrus.Debugf("About to force-kill %s\n", operation)
-	err := killFunc()
-	if err != nil {
-		return err
-	}
-	return nil
+	return killFunc()
 }
 
 /**
- * checkProcessX function returns true if it detects the app is still running, false otherwise
+ * checkProcessX function returns [true, nil] if it detects the app is still running, [false, X] otherwise
+ * Linux/macOS errors are always nil, Windows not so
```

Comment:
So functionally, the boolean return is always redundant: the error return is either nil (it is running) or non-nil (it is no longer running)? That sounds incorrect; maybe reword this?

Reply:
Not exactly. I reworded the comment. All I wanted was an error-situation in `factoryreset.CheckProcessWindows` to be propagated.